src/Controller/WopiController.php

```diff
@@ -148,28 +148,71 @@ protected function wopiPutFile(MediaInterface $media, FileInterface $file, UserI
     }
 
     $new_file_content = $request->getContent();
-    $new_file = $this->createNewFileEntity($file, $new_file_content);
-
+    $new_file_interval = $this->configFactory->get('collabora_online.settings')->get('cool.new_file_interval') ?? 0;
+    $request_time = $this->time->getRequestTime();
     $save_reason = $this->buildSaveReason($request);
 
+    if (
+      $new_file_interval === 0 ||
+      $request_time - $file->getCreatedTime() <= $new_file_interval
+    ) {
+      // Replace file with new content.
+      $this->fileSystem->saveData(
+        $new_file_content,
+        $file->getFileUri(),
+        FileExists::Replace,
+      );
+
+      // Entity didn't change but file has been replaced.
+      $file->setChangedTime($request_time);
+      $file->save();
+
+      $this->logger->info(
+        'The file @file_uri was replaced with Collabora.<br>
+Save reason: @reason<br>
+File ID: @file_id<br>
+Media ID: @media_id<br>
+User ID: @user_id',
+        [
+          '@file_uri' => $file->getFileUri(),
+          '@reason' => $save_reason,
+          '@file_id' => $file->id(),
+          '@media_id' => $media->id(),
+          '@user_id' => $user->id(),
+        ],
+      );
```

Comment:
Just a thought, but not a hard blocker:
Imo the media id (or title) is the thing you want to see in the first line, instead of the file path.
And we could say "overwritten" instead of "replaced".

E.g.
```suggestion
      $this->logger->info(
        'The file contents for media @media_id were overwritten with Collabora.<br>
Save reason: @reason<br>
File: @file_id / @file_uri<br>
User ID: @user_id',
        [
          '@media_id' => $media->id(),
          '@file_uri' => $file->getFileUri(),
          '@reason' => $save_reason,
          '@file_id' => $file->id(),
          '@user_id' => $user->id(),
        ],
      );
```

Reply:
(We might find more ways to improve these log messages in the future, this is why I say not a blocker, the message already contains everything it needs.)

Reply:
Changed, the suggestion it's ok and seems more standardised.